### PR TITLE
Faster decoding related changes

### PR DESCRIPTION
**Pull request by rajatsen91**
1. Fused QKV (caution: need to update the weights on huggingface)
2. Updates decode cache for the whole batch at once.
3. Use torch's fused attention kernel.
4. Torch compile related suggestions.

Still TODO:

1. Make ROPE faster or less redundant

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,8 +57,12 @@ pip install -e .
 ### Code Example
 
 ```python
+import torch
 import numpy as np
 import timesfm
+
+torch.set_float32_matmul_precision("high")
```
Comment (siriuz42) [inline README.md:64]: This might affect the GIFT-Eval numbers we reported. Let's try another run when we've got time?

File: src/timesfm/timesfm_2p5/timesfm_2p5_base.py [hunk 2]
```diff
@@ -85,43 +87,94 @@ class TimesFM_2p5_200M_Definition:
...
-      output_dims=10240,
+      num_heads=16,
+      attention_norm="rms",
+      feedforward_norm="rms",
+      qk_norm="rms",
       use_bias=False,
-      activation="swish",
+      use_rotary_position_embeddings=True,
+      ff_activation="swish",
+      fuse_qkv=True,
```
[the remark this answers is not in the document]
Comment (rajatsen91) [reply src/timesfm/timesfm_2p5/timesfm_2p5_base.py:162]: I am ok with that.

File: src/timesfm/torch/transformer.py [hunk 3]
```diff
@@ -114,23 +112,45 @@ def forward(
...
     )
 
   attn_weights = F.softmax(attn_weights, dim=-1)
 
   return torch.einsum("...hqk,...khd->...qhd", attn_weights, value)
 
 
+def _torch_dot_product_attention(query, key, value, mask=None):
+  """
+  Performs the exact same (unscaled) attention as your original function,
```
Comment (siriuz42) [inline src/timesfm/torch/transformer.py:134]: `your`
